src/uwtools/cli.py

```diff
@@ -22,28 +24,32 @@
 FORMATS = [FORMAT.ini, FORMAT.nml, FORMAT.yaml]
 TITLE_REQ_ARG = "Required arguments"
 
+SubmodeChecks = List[Callable[[Namespace], Namespace]]
+ModeChecks = Dict[str, SubmodeChecks]
+Checks = Dict[str, ModeChecks]
+
 
 def main() -> None:
     """
     Main entry point.
     """
 
-    # Silence logging initially, then process the command-line arguments by parsing them, filling
-    # in any unspecified data-format arguments, and checking semantic argument validity (i.e. that
-    # the arguments make sense together, not just on their own). If the arguments are sane, set up
-    # logging correctly, then dispatch to the mode handler, which will the dispatch to the submode
-    # handler. Shield command-line users from raised exceptions by aborting gracefully.
+    # Silence logging initially, then process the command-line arguments by parsing them. Run all
+    # defined checks for the appropriate [sub]mode. Reconfigure logging after quiet/verbose choices
+    # are known, then dispatch to the [sub]mode handler.
 
     setup_logging(quiet=True)
-    modes = {
-        "config": _dispatch_config,
-        "forecast": _dispatch_forecast,
-        "template": _dispatch_template,
-    }
     try:
-        args = _check_args(_set_formats(_parse_args(sys.argv[1:])))
+        args, checks = _parse_args(sys.argv[1:])
+        for check in checks[args.mode][args.submode]:
+            check(args)
         setup_logging(quiet=args.quiet, verbose=args.verbose)
         logging.debug("Command: %s %s", Path(sys.argv[0]).name, " ".join(sys.argv[1:]))
+        modes = {
+            STR.config: _dispatch_config,
+            STR.forecast: _dispatch_forecast,
+            STR.template: _dispatch_template,
+        }
```

Comment:
This block has the first of many references to the `STR` object to obtain string values. Just in the course of preparing this PR, my editor was able to catch a number of would-be typos, e.g. `STR.tempate` was flagged immediately by my editor as an error, while there's nothing it could say about `"tempate"`.